source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,156 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding only a subset of them in memory at a
+given time. Read operations that match multiple documents use a cursor
+to return those documents in batches as opposed to all at once.
+
+Sample Cursor
+~~~~~~~~~~~~~
+
+Each section uses the following ``cursor`` variable, which is a
+``Cursor`` struct that contains all the documents in a collection:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+.. _individual-documents-golang:
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from your cursor individually, use the
+``Next()`` method.
+
+This method blocks the current goroutine. That means the function
+returns a document if each of the following is true:
+
+- A document is currently or will later be available
+- No errors occurred
+- The ``MongoClient`` didn't exhaust the cursor
+- The context didn't expire
+
+.. include:: /includes/fundamentals/exhausted-cursor-definition.rst
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin cursor next
+   :end-before: end cursor next
+
+Documents Only in the Cursor
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
[Responding to this comment thread](https://github.com/mongodb/docs-golang/pull/79#discussion_r746946026) as I it seems like I can't respond through the github UI anymore.

The [tailable cursors](https://docs.mongodb.com/manual/core/tailable-cursors/) page says the following: 

> However, for capped collections you may use a Tailable Cursor that remains open after the client exhausts the results in the initial cursor.

I agree this wording is unclear as to whether tailable cursors are only available in capped collections. Could a ticket be filed with server team to clarify whether or not cursors are only available for capped collections?

The Go [API docs say](https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.2/mongo#Cursor.TryNext):

> This is only recommended for use with tailable cursors as a non-blocking alternative to Next

Could this section be titled to better reflect this use case? I'm not sure if this is an equivalent condition as "Documents only being in the Cursor". Could the title be changed to reflect "blocking" vs "non-blocking"?